integration-tests/zendeskRequestAPI.spec.ts

```diff
@@ -21,30 +94,39 @@ describe('Zendesk request integrity', () => {
     }
 
     const headers = {
-      'X-Zendesk-Webhook-Signature': invalidSignature
+      ...generateSignatureHeaders(webhookRequestData)
     }
 
-    const invalidRequestError = async () => {
-      return axios({
-        url: getEnvVariable('AWS_WEBHOOK_URL'),
-        method: 'POST',
-        headers,
-        data
-      })
-        .then(() => {
-          throw 'Gateway did not return an error'
-        })
-        .catch((error) => {
-          if (error.response) {
-            return error.response
-          }
-          console.log(error)
-          throw 'No response recieved from gateway'
-        })
+    const errorResponse = await invalidRequestError(headers, webhookRequestData)
+    expect(errorResponse.status).toEqual(404)
+    expect(errorResponse.data.message).toEqual('Zendesk ticket not found')
+  })
+
+  test('API Gateway returns a 400 response if the request does not match info in corresponding Zendesk ticket', async () => {
+    const webhookRequestData = {
+      //ticket with ID 881 exists with different info
+      zendeskId: '881',
+      resultsEmail: 'user@test.gov.uk',
+      resultsName: 'test user',
+      dateFrom: '2022-09-06',
+      dateTo: '2022-09-06',
+      identifierType: 'journey_id',
+      sessionIds: '',
+      journeyIds: '098 565 2213',
+      userIds: '',
+      eventIds: '',
+      piiTypes: 'name',
+      dataPaths: ''
+    }
+
+    const headers = {
+      ...generateSignatureHeaders(webhookRequestData)
     }
 
-    const errorResponse = await invalidRequestError()
-    expect(errorResponse.status).toBe(400)
-    expect(errorResponse.data.message).toBe('Invalid request source')
+    const errorResponse = await invalidRequestError(headers, webhookRequestData)
+    expect(errorResponse.status).toEqual(400)
```

Comment:
Can you add a check for the log error as well as per the a/c? There are now helper functions for that.

Reply:
> This does the same thing though? Essentially we'd be calling AWS to look in the logs for this very message!

Same here!